catalogs/io/v5_0.py

```diff
@@ -0,0 +1,193 @@
+    def implementation(self) -> str:
+        part = self._get_part("implementation")
+
+        if not part:
+            return ""
+
+        return part.prose
+
+    @property
+    def guidance(self) -> str:
+        part = self._get_part("guidance")
+
+        if not part:
+            return ""
+
+        return part.prose
+
+    @property
+    def description(self) -> str:
+        def _get_prose(item, depth=0):
+            depth += 1
+            tabs = "\t" * depth
+
+            if prose := getattr(item, "prose", ""):
+                prose = f"\n{tabs}{item.label}. {prose}"
+                parts.append(prose)
+
+            if parts_ := getattr(item, "parts", []):
+                for part in parts_:
+                    _get_prose(part, depth)
+
+        parts = []
+        _get_prose(self.statement, depth=-3)
+
+        return "".join(parts).strip()
+
+    def to_orm(self) -> dict:
+        return {
+            "control_id": self.id,
+            "control_label": self.label,
+            "sort_id": self.sort_id,
+            "title": self.title
+        }
+
+
+class Family(BaseControl):
+    item_class: Literal["family"] = Field(..., alias="class")
+    controls: list[Control]
+
+    @validator("controls")
+    def set_family_id(cls, value: list[Control], values) -> list[Control]:  # pylint: disable=no-self-argument
+        for item in value:
+            if not item.family_id:
+                item.family_id = values.get("id")
+
+        return value
+
+
+class CatalogModel(BaseModel):
+    uuid: UUID4
+    metadata: CatalogMeta
+    groups: list[Family]
+
+    @property
+    def controls(self) -> list[Control]:
+        return sorted((item for group in self.groups for item in group.controls), key=lambda control: control.sort_id)
+
+    def get_control(self, control_id: str) -> Optional[Control]:
+        return next(filter(lambda control: control.id == control_id, self.controls), None)
+
+    def get_next(self, control: Control) -> str:
+        try:
+            next_idx = self.controls.index(control) + 1
+            return self.controls[next_idx].id
+        except (ValueError, IndexError):
+            return ""
+
+    def control_summary(self, control_id: str) -> dict:
+        control = self.get_control(control_id)
+        next_id = self.get_next(control)
+
+        return {
+            "label": control.label,
+            "sort_id": control.sort_id,
+            "title": control.title,
+            "family": control.family_id,
+            "description": control.description,
+            "implementation": control.implementation,
+            "guidance": control.guidance,
+            "next_id": next_id
+
+        }
+
+    @classmethod
+    def from_json(cls, json_file: Union[str, Path]):
+        with open(json_file, "rb") as file:
+            data = json.load(file)
+
+        try:
+            return cls(**data)
+        except ValidationError:  # Try nested "catalog" field
+            return cls(**data["catalog"])
```

Comment:
Talked about this a bit more "in person". The idea of consolidating "io" code with the catalog, component, etc repos is a great idea, but outside the scope of blueprint. In the future, blueprint could use that rather than implementing its own parsing code and schemas. 

Consolidating the models will happen in a separate, followup PR.